Sources/NIO/SelectableEventLoop.swift

```diff
@@ -87,20 +87,20 @@ internal final class SelectableEventLoop: EventLoop {
     /// Creates a new `SelectableEventLoop` instance that is tied to the given `pthread_t`.
 
     private let promiseCreationStoreLock = Lock()
-    private var _promiseCreationStore: [ObjectIdentifier: (file: StaticString, line: UInt)] = [:]
+    private var _promiseCreationStore: [UInt: (file: StaticString, line: UInt)] = [:]
 
     @usableFromInline
     internal func promiseCreationStoreAdd<T>(future: EventLoopFuture<T>, file: StaticString, line: UInt) {
         precondition(_isDebugAssertConfiguration())
         self.promiseCreationStoreLock.withLock {
-            self._promiseCreationStore[ObjectIdentifier(future)] = (file: file, line: line)
+            self._promiseCreationStore[UInt(bitPattern: ObjectIdentifier(future))] = (file: file, line: line)
```

Comment:
I think we should have a

```swift
private func obfuscatePointerValue<T>(_ future: EventLoopFuture<T>) -> UInt {
    return UInt(bitPattern: ObjectIdentifier(future)) ^ 0xbf15ca5d
}
```

to convert the pointers to `UInt`s. Also, we should XOR it with a value (I don't mind which value we use, the above proposes `0xbf15ca5d` which reads a little bit like _obfuscated_) so that it's hidden from Memory Graph debugger which always (incorrectly) claims that there's a reference cycle :)